service/switchboardapp/src/main/java/com/switchboard/app/entity/StreamEntity.java

```diff
@@ -0,0 +1,28 @@
+package com.switchboard.app.entity;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import lombok.*;
+
+import javax.persistence.*;
+import java.util.Base64;
+
+@Entity(name = "Stream")
+@Data
+@JsonIgnoreProperties({"hibernateLazyIntializer", "handler"})
+@AllArgsConstructor
+@Builder
+public class StreamEntity {
+
+    @Id
+    @GeneratedValue
+    @Column(name = "id")
+    private long id;
+
+    @ManyToOne
+    @JoinColumn(name = "encoder_id", referencedColumnName = "serial_number")
+    private EncoderEntity encoder;
+
+    @ManyToOne
+    @JoinColumn(name = "decoder_id", referencedColumnName = "serial_number")
+    private DecoderEntity decoder;
```

Comment:
if you look at our ER diagram a stream is associated with an encoder and a channel.
That means that to initiate a stream you would need an encoderID, port.
In the ER diagram built it is a many to many relationship between encoder and channel.
Actually for the purpose of our application i dont think we need a new entity at all.

Reply:
I think all we need is crud operations for the channel entity and to initiate a stream. You would have a manytomany relation from channel to encoder (which we already have on master) but add an attribute called vid which is the id of the video being streamed